### PR TITLE
15 signup

### PR DESCRIPTION
**Pull request by Eihcir0**
This pr implements the new Volunteer Signup flow in accordance with [card #15](https://www.figma.com/file/v5HdxDsvO2NzMuZU6CArzF/Resilience-App?node-id=1841%3A2718)

The design doesn't match perfectly.  I spent some time trying to clean it up, but also trying to use our native styled components as much as possible.  Notable differences include: font sizing, input selector placeholder text styling, and also FYI on the final page I used our `SuccessSnackbar` instead of the informational alert shown at the top of the page on the design.

Also I had to steal @pedro-surf 's awesome AddressInput for this as it has yet to be merged

@mat10tng @utunga FYI: The schema for `user` that I saw  on codeshare.io did not include the `availability` or `description` (tell us about yourself) fields seen on the new version of the signup form

**TODO**:
- form validation
- submit final payload to firebase
- TESTS

As for the code, I do not use material-ui or react with Hooks at my day job so trying to learn as I go here.  I love feedback so please let me know any ways I can improve.

### CODE REVIEW

**Comment by Eihcir0** [comment]
Screenshots:
Screen Shot 2020-04-14 at 3 30 24 PM
Screen Shot 2020-04-14 at 3 31 14 PM
Screen Shot 2020-04-14 at 12 05 18 PM
Screen Shot 2020-04-14 at 12 05 30 PM
Screen Shot 2020-04-14 at 12 06 41 PM
Screen Shot 2020-04-14 at 12 19 18 PM

File: src/app/page/Signup/SignupScene.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import PropTypes from 'prop-types';
+import useForm from "../../hooks/useForm";
+
+import CallToActionPage from './CallToAction';
+import UserProfilePage from "./UserProfile";
+import ConnectSocialMediaPage from "./ConnectSocialMedia";
+import PhoneAuthPage from "./PhoneAuth";
+import SignupSuccessPage from "./SignupSuccess";
+
+
```
Comment (azmainamin) [inline src/app/page/Signup/SignupScene.jsx:12]: Can we add some jsdoc to this function?

File: src/app/page/Signup/SignupScene.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      const parts = fullName.split(' ')
+      lastName = parts[parts.length - 1]
+      if (parts.length > 1) {
+        firstName = parts.slice(0, parts.length - 1)
+      }
+    }
+    return [firstName, lastName]
+  };
+
+
```
Comment (azmainamin) [inline src/app/page/Signup/SignupScene.jsx:26]: Can we run prettier on this?

File: src/app/page/Signup/SignupScene.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ *
+ */
+function SignupScene(props) {
+    const { handleChange, values } = useForm();
+    const [activeTab, setActiveTab] = useState(0);
+
+    function changeFormValues(changes) {
+        for (const change of changes) {
+            const [name, value] = change
+            console.log('setting' + name + ' to ' + value)
```
Comment (azmainamin) [inline src/app/page/Signup/SignupScene.jsx:39]: Remove log.

**Comment by azmainamin** [comment]
I also could not run your code because of the conflict. Can you resolve them? Then I'd like to run it and do a quick manual check.

File: src/app/component/AddressInput/AddressInput.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      });
+    }
+  };
+  return (
+    <AlgoliaPlaces
+      placeholder={placeholder || "Search address"}
+      name={stage}
+      options={{
+        appId: "plZ318O8ODTC",
+        apiKey: "b5e0781d289a9aa8edb37bf24aef874e", // only 1k per day
```
Comment (qdozaq) [inline src/app/component/AddressInput/AddressInput.jsx:28]: Can we abstract this out into a constant somewhere? I know theres a few other places this is used.

File: src/app/component/MapView/MapView.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { Map, TileLayer, Marker, Popup } from "react-leaflet";
 import { MapStyle } from "./MapView.style";
 
 function MapView(props) {
-  const [state, setState] = React.useState({
+  const [state] = React.useState({
```
[the remark this answers is not in the document]
Comment (qdozaq) [reply src/app/component/MapView/MapView.js:8]: Yeh I don't think we need state for this just pass the props directly to `position`

File: src/app/page/MakeMission/MakeMission.jsx [hunk 6]
```diff
@@ -15,7 +15,7 @@ function MakeMission({ history, firestore }) {
   const [file, setFile] = useState(null);
   // const [autoAssignHelper, setAutoAssignHelper] = useState(false);
   const storage = firebase.storage();
-  const { handleChange, values, setValues } = useForm();
```
[the remark this answers is not in the document]
Comment (qdozaq) [reply src/app/page/MakeMission/MakeMission.jsx:18]: I think thats only to set the values manually. otherwise `handleChange` takes care of it

**Comment by azmainamin** [comment]
I super agree with Paul about keeping PRs small and manageable. It took me a lot of time going through all of this.

File: src/app/page/Login/Login.jsx [hunk 7]
```diff
@@ -43,15 +43,16 @@ const LoginPage = (props) => {
     );
   } else {
     if (firstTimeSignIn(auth)) {
+      return <Redirect to="/signup"/>;
```
[the remark this answers is not in the document]
Comment (Eihcir0) [reply src/app/page/Login/Login.jsx]: This whole page probably needs to be revisited but for now am reverting to the original as its not part of this card

File: src/app/component/MapView/MapView.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Eihcir0) [reply src/app/component/MapView/MapView.js]: Reverted this change and all extranneous cleanup as its not relevant to this pr

File: src/app/page/Login/Login.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Eihcir0) [reply src/app/page/Login/Login.jsx]: Reverted all changes to this page as its not relevant to this pr

**Comment by Eihcir0** [comment]
@azmainamin  @qdozaq 
Thanks for the review.

I've made all requested changes, resolved the merge conflicts, and run prettier on everything.

I've reverted any code changes (i.e. some of the 'cleanup' items you commented on) as they aren't relevant to this pr.

Please let me know if there's anything else I can do to make this better

**Comment by Eihcir0** [comment]
I've made all the requested changes, resolved the merge conflicts, and run prettier on everything.
I've reverted any code changes (i.e. some of the 'cleanup' items you commented on) as they aren't relevant to this pr

**Pull request merged**